tests/directory_parser/test_directory_parser.py

```diff
@@ -9,59 +9,56 @@
-
-        # Get the contents of the folder - dirs and files
-        folder_path = urljoin(mhttpd.HERE, mhttpd.WDIR, 'directoryparser',
-                              'filter')
-        contents = os.listdir(folder_path)
-        contents.sort()
-        self.assertEqual(parser.entries, contents)
-
-        # filter out files
-        parser.entries = parser.filter(r'^\d+$')
-
-        # Get only the subdirectories of the folder
-        dirs = os.walk(folder_path).next()[1]
-        dirs.sort()
-        self.assertEqual(parser.entries, dirs)
-
-        # Test filter method with a function
-        parser.entries = parser.filter(lambda x: x == dirs[0])
-        self.assertEqual(parser.entries, [dirs[0]])
-
-    def test_names_with_spaces(self):
-        parser = DirectoryParser(urljoin(self.wdir, 'directoryparser', 'some spaces/'))
-
-        # Get the contents of the folder - dirs and files
-        folder_path = urljoin(mhttpd.HERE, mhttpd.WDIR, 'directoryparser',
-                              'some spaces')
-        contents = os.listdir(folder_path)
-        contents.sort()
-        self.assertEqual(parser.entries, contents)
+
+def test_init(httpd):
+    """Testing the basic functionality of the DirectoryParser Class"""
+
+    # DirectoryParser returns output
+    parser = DirectoryParser(httpd.get_url())
+
+    # relies on the presence of other files in the directory
+    # Checks if DirectoryParser lists the server entries
+    assert parser.entries != [], "parser.entries were not listed"
+
+    # path_regex to mozdownload -t release -p win32 -v latest
+    testpath = urljoin(httpd.get_url(), 'directoryparser/')
+    parser1 = DirectoryParser(testpath)
+    parser1.entries.sort()
+    testdir = os.listdir(urljoin(httpd.router.doc_root, 'directoryparser'))
+    testdir.sort()
+    assert parser1.entries == testdir
+
+
+def test_filter(httpd):
+    """Testing the DirectoryParser filter method"""
+
+    parser = DirectoryParser(urljoin(httpd.get_url(), 'directoryparser', 'filter/'))
+
+    # Get the contents of the folder - dirs and files
+    folder_path = urljoin(httpd.router.doc_root, 'directoryparser', 'filter')
+    contents = os.listdir(folder_path)
+    contents.sort()
+    assert parser.entries == contents
+
+    # filter out files
+    parser.entries = parser.filter(r'^\d+$')
+
+    # Get only the subdirectories of the folder
+    dirs = os.walk(folder_path).next()[1]
+    dirs.sort()
+    assert parser.entries == dirs
+
+    # Test filter method with a function
+    parser.entries = parser.filter(lambda x: x == dirs[0])
+    assert parser.entries == [dirs[0]]
+
+
+def test_names_with_spaces(httpd):
+
+    parser = DirectoryParser(urljoin(httpd.get_url(), 'directoryparser', 'some spaces/'))
+
+    # Get the contents of the folder - dirs and files
+    folder_path = urljoin(httpd.router.doc_root, 'directoryparser', 'some spaces')
+    contents = os.listdir(folder_path)
+    contents.sort()
+    assert parser.entries == contents
```

Comment:
Same here as above for sorting parser entries.

Reply:
referenced in the above comments.